slsDetectorSoftware/src/DetectorImpl.cpp

```diff
@@ -1281,6 +1303,58 @@ void DetectorImpl::startAcquisition(bool blocking, std::vector<int> positions) {
     }
 }
 
+void DetectorImpl::sendSoftwareTrigger(const bool block,
+                                       std::vector<int> positions) {
+    bool handleSynchronization = false;
+    // multi module sync enabled jungfrau
+    if (size() > 1 && (shm()->detType == defs::JUNGFRAU) &&
+        (Parallel(&Module::getSynchronizationFromStopServer, positions)
+             .tsquash("Inconsistent synchronization among modules"))) {
+        handleSynchronization = true;
+    }
+
+    // slaves first
+    if (handleSynchronization) {
+        std::vector<int> master;
+        std::vector<int> slaves;
+        getMasterSlaveList(positions, master, slaves);
+        if (!slaves.empty())
+            Parallel(&Module::sendSoftwareTrigger, slaves, false);
+        if (!master.empty())
+            Parallel(&Module::sendSoftwareTrigger, master, block);
+    }
+    // all in parallel
+    else {
+        Parallel(&Module::sendSoftwareTrigger, positions, block);
+    }
+}
+
+void DetectorImpl::stopDetector(std::vector<int> positions) {
```

Comment:
stop only has special case for Jungfrau not M3?

Reply:
Ill add M3 this time and G2 and G1 in another PR after what the next meeting entails.

Reply:
done